v2x_ros_driver/src/v2x_radio_client.cpp

```diff
@@ -168,29 +169,138 @@ void V2XRadioClient::process(const std::shared_ptr<const std::vector<uint8_t>>&
+                }
+            }
         }
     }
 }
 
+void V2XRadioClient::printVector(const std::vector<uint8_t>& vec)
+{
+    std::ostringstream oss;
+    oss << "[";
+    for (size_t i = 0; i < vec.size(); ++i) {
+        oss << static_cast<int>(vec[i]);
+        if (i < vec.size() - 1) {
+            oss << ", ";
+        }
+    }
+    oss << "]";
+    RCLCPP_INFO_STREAM(logger_, "Sending message vector: " << oss.str());
+}
+
+std::vector<uint8_t> V2XRadioClient::hexStringToByteArray(const std::string& hexString) const
+{
+    std::vector<uint8_t> byteArray;
+
+    // Loop through the hex string, two characters at a time
+    for (size_t i = 0; i < hexString.length(); i += 2) 
+    {
+        // Extract two characters representing a byte
+        auto byteString = hexString.substr(i, 2);
+
+        // Convert the byte string to a uint8_t value
+        auto byteValue = static_cast<uint8_t>(stoi(byteString, nullptr, 16));
+
+        // Add the byte to the byte array
+        byteArray.push_back(byteValue);
+    }
+
+    return byteArray;
+}
+
+bool V2XRadioClient::isValidMsgSize(const std::vector<uint8_t> msg_vec, size_t start_index, size_t end_index, const std::vector<uint8_t> entry)
+{
+    if (msg_vec.size() > 255) 
+    {
+        auto tmp_start_index = start_index + long_frame_;
+        std::vector<uint8_t> long_vec(entry.begin() + tmp_start_index, entry.begin() + end_index);
+        auto msg_size = (static_cast<uint16_t>(msg_vec[2]) << 8) | msg_vec[3];
+        if (msg_size == long_vec.size())
+            {
+                return true;
+            }
+        else 
+        {
+            RCLCPP_WARN_STREAM(logger_, "Size in possible MessageFrame does not match actual data size. Checking rest of data.");
+            return false;
+        }
+    }
+    else 
+    {
+        auto tmp_start_index = start_index + short_frame_;
+        std::vector<uint8_t> short_vec(entry.begin() + tmp_start_index, entry.begin() + end_index);
+        auto msg_size = msg_vec[2];
+        if (msg_size == short_vec.size())
+            {
+                return true;
+            }
+        else {
+            RCLCPP_WARN_STREAM(logger_, "Size in possible MessageFrame does not match actual data size. Checking rest of data.");
+            return false;
+        }
+    }
+}
+
+bool V2XRadioClient::isValidPSID(const std::string &msg_id)
+{
+    if (this->wave_cfg_psids_.empty())
+    {
+        if (this->wave_file_path.empty())
+        {
+            return false;
+        }
+        loadWaveConfigIds(this->wave_file_path);
+    }
+
+    for (const auto &psid : this->wave_cfg_psids_) 
+    {
+        // Convert the entire hex string to an integer
+        int psid_value = std::stoi(psid, nullptr, 16);
+        // Convert the integer to its decimal string representation
+        std::string psidInt = std::to_string(psid_value);
+        if (msg_id == psidInt)
+            RCLCPP_WARN_STREAM(logger_, "PSID found, parsing rest of data for MessageID.");
```

Comment:
I think we need to move all or part of this log to the calling function. This log states "parsing rest of data..." but `isValidPSID()` does not do anymore checks, they're handled in `process()`

Reply:
Good call, updated